include/deal.II/hp/fe_collection.h

```diff
@@ -450,10 +450,35 @@ namespace hp
 
   /* --------------- inline functions ------------------- */
 
+  namespace
+  {
+    template<bool... Types> struct BoolStorage;
+
+    template<class U, class... Types>
+    struct is_base_of_all
+    {
+      static constexpr bool value =
+        std::is_same<BoolStorage<std::is_base_of<U,Types>::value..., true>,
+        BoolStorage<true, std::is_base_of<U,Types>::value...>>::value;
```

Comment:
Sure, I can move this. Probably, we want to have something different than `is_base_of` on other places but it's clearly not bad to have this pattern in a central place.